services/wallet/concurrent.go

```diff
@@ -45,17 +45,56 @@ type TransferDownloader interface {
 	GetTransfersByNumber(context.Context, *big.Int) ([]Transfer, error)
 }
 
+var balancesCache = map[common.Address]map[*big.Int]*big.Int{}
+var balanaceMapLock = sync.RWMutex{}
+
+func readCachedBalance(account common.Address, blockNumber *big.Int) (*big.Int, bool) {
+	balanaceMapLock.RLock()
+	defer balanaceMapLock.RUnlock()
+
+	balance, exists := balancesCache[account][blockNumber]
```

Comment:
```suggestion
	return balancesCache[account][blockNumber]
```

Reply:
if I do so how do I check whether map contains the value? because I also have to change signature of the function and it wouldn't return that `ok` value or something